src/isomorphic/modern/class/ReactBaseClasses.js

```diff
@@ -27,6 +27,9 @@ function ReactComponent(props, context, updater) {
   this.refs = emptyObject;
   // We initialize the default updater but the real one gets injected by the
   // renderer.
+  if (this.updater && __DEV__) {
+      warning(this instanceof Constructor, 'This object has a property called updater which will be overwritten by React default updater');
```

Comment:
The plan is to display a warning whenever someone is trying to update the `updater` property, just like in your example. 

I was under the impression that checking if `this.updater` is perhaps already defined and if it is and we're in `dev` mode should cause a warning to be displayed here.